### PR TITLE
Cleanup and move char functions to the `char_utils.h` header.

### PR DESCRIPTION
**Pull request by bruvzg**
Cleanup, deduplicate and move char related function to the new `char_utils.h` header.

### CODE REVIEW

File: core/io/json.cpp [hunk 1]
```diff
@@ -335,10 +335,10 @@ Error JSON::_get_token(const char32_t *p_str, int &index, int p_len, Token &r_to
 					r_token.value = number;
 					return OK;
 
-				} else if ((p_str[index] >= 'A' && p_str[index] <= 'Z') || (p_str[index] >= 'a' && p_str[index] <= 'z')) {
+				} else if (is_acsii_char(p_str[index])) {
```
Comment (akien-mga) [inline core/io/json.cpp:338]: Typo (`aCSii`)

File: core/io/resource_uid.cpp [hunk 2]
```diff
@@ -71,9 +71,9 @@ ResourceUID::ID ResourceUID::text_to_id(const String &p_text) const {
 	for (uint32_t i = 6; i < l; i++) {
 		uid *= base;
 		uint32_t c = p_text[i];
-		if (c >= 'a' && c <= 'z') {
+		if (is_acsii_lower_case(c)) {
```
Comment (akien-mga) [inline core/io/resource_uid.cpp:74]: Typo.

File: editor/find_in_files.cpp [hunk 3]
```diff
@@ -73,10 +68,10 @@ static bool find_next(const String &line, String pattern, int from, bool match_c
 		out_end = end;
 
 		if (whole_words) {
-			if (begin > 0 && is_text_char(line[begin - 1])) {
+			if (begin > 0 && (is_ascii_identifier_char(line[begin - 1]))) {
 				continue;
 			}
-			if (end < line.size() && is_text_char(line[end])) {
+			if (end < line.size() && (is_ascii_identifier_char(line[end]))) {
```
Comment (akien-mga) [inline editor/find_in_files.cpp:74]: Strictly speaking the parentheses are redundant here (but that's not a big deal :)).

Comment (akien-mga) [reply editor/find_in_files.cpp:74]: Yeah don't bother, it's fine as is :)
There's other places where it was already using this style.

**Review by akien-mga — APPROVED** [review]
Excellent work! Reviewed the whole diff and it seems all good to me.

**Comment by akien-mga** [comment]
Thanks!

**Comment by belzecue** [comment]
Apologies for a silly question -- I'm not a C++ programmer.  But why repeat definitions instead of using already defined ones? e.g. is_ascii_alphanumeric_char is:

```
static _FORCE_INLINE_ bool is_ascii_alphanumeric_char(char32_t c) {
	return (c >= 'a' && c <= 'z') || (c >= 'A' && c <= 'Z') || (c >= '0' && c <= '9');
}

```

But why not:

```
static _FORCE_INLINE_ bool is_ascii_alphanumeric_char(char32_t c) {
	return is_ascii_char(c) || is_digit(c);
}

```
since those are functions defined earlier in the file?

**Pull request merged**